Automattic/wp-calypso
# Add a theme collection component for TS and support for Premium and Partner collections

This PR introduces a new ThemeCollection component that will group various types of themes (e.g. Premium, Partner). 

The implementation is currently behind a feature flag since the implementation is not complete: we have to properly style it and add an additional layout component that will display collections and banners, for example. 

Related to https://github.com/Automattic/dotcom-forge/issues/3959, https://github.com/Automattic/dotcom-forge/issues/3989

## Proposed Changes

* Add a generic component for Themeshowcase that displays the themes in a swiper carousel
* Add two components for Premium and Partner themes
* The implementation is behind a flag and not visible to users

## Testing Instructions
- Use the Calypso live link and go to /themes page in incognito.
- Append the feature flag `flags=themes/discovery-lots`
- Notice that we display three carousels:
- - Premium themes 
- - Partner themes 
- Ignore the unstyled carousel, which will be handled in a separate PR.

![Screenshot 2023-10-06 at 17 27 40](https://github.com/Automattic/wp-calypso/assets/995955/daf4f896-ebe6-4090-aaf7-ce7f1f699a18)


## Pre-merge Checklist
